core/src/oauth/providers/gong.rs

```diff
@@ -0,0 +1,161 @@
+    static ref OAUTH_GONG_CLIENT_SECRET: String = env::var("OAUTH_GONG_CLIENT_SECRET").unwrap();
+}
+
+pub struct GongConnectionProvider {}
+
+impl GongConnectionProvider {
+    pub fn new() -> Self {
+        GongConnectionProvider {}
+    }
+}
+
+#[async_trait]
+impl Provider for GongConnectionProvider {
+    fn id(&self) -> ConnectionProvider {
+        ConnectionProvider::Gong
+    }
+
+    async fn finalize(
+        &self,
+        _connection: &Connection,
+        code: &str,
+        redirect_uri: &str,
+    ) -> Result<FinalizeResult, ProviderError> {
+        let authorization = format!(
+            "Basic
+        {}",
+            STANDARD.encode(format!(
+                "{}:{}",
+                *OAUTH_GONG_CLIENT_ID, *OAUTH_GONG_CLIENT_SECRET
+            ))
+        );
+        let params = [
+            ("grant_type", "authorization_code"),
+            ("code", &code),
+            ("redirect_uri", &redirect_uri),
+        ];
+
+        let req = reqwest::Client::new()
+            .post("https://app.gong.io/oauth2/generate-customer-token")
+            .header("Content-Type", "application/json")
+            .header("Authorization", authorization)
+            .query(&params);
+
+        let raw_json = execute_request(ConnectionProvider::Gong, req)
+            .await
+            .map_err(|e| self.handle_provider_request_error(e))?;
+
+        let access_token = match raw_json["access_token"].as_str() {
+            Some(token) => token,
+            None => Err(anyhow!("Missing `access_token` in response from Gong"))?,
+        };
+
+        let expires_in = match raw_json.get("expires_in") {
+            Some(serde_json::Value::Number(n)) => match n.as_u64() {
+                Some(n) => n,
+                None => Err(anyhow!("Invalid `expires_in` in response from Gong"))?,
+            },
+            _ => Err(anyhow!("Missing `expires_in` in response from Gong"))?,
+        };
+
+        let refresh_token = match raw_json["refresh_token"].as_str() {
+            Some(token) => token,
+            None => Err(anyhow!("Missing `refresh_token` in response from Gong"))?,
+        };
+
+        Ok(FinalizeResult {
+            redirect_uri: redirect_uri.to_string(),
+            code: code.to_string(),
+            access_token: access_token.to_string(),
+            access_token_expiry: Some(
+                utils::now() + (expires_in - PROVIDER_TIMEOUT_SECONDS) * 1000,
+            ),
+            refresh_token: Some(refresh_token.to_string()),
+            raw_json,
+        })
+    }
+
+    async fn refresh(&self, connection: &Connection) -> Result<RefreshResult, ProviderError> {
+        let refresh_token = match connection.unseal_refresh_token() {
+            Ok(Some(token)) => token,
+            Ok(None) => Err(anyhow!("Missing `refresh_token` in Gong connection"))?,
+            Err(e) => Err(e)?,
+        };
+
+        let body = json!({
```

Comment:
Based on the documentation, this should be done using query parameters no body and only `grant_type` and `refresh_token` are required. The Authorization header should be set like you do in `finalize`.